bootstrap/eks/controllers/eksconfig_controller.go

```diff
@@ -0,0 +1,234 @@
+			log.Info(fmt.Sprintf("%s does not belong to a cluster yet, requeueing until it's part of a cluster", configOwner.GetKind()))
+			return ctrl.Result{}, nil
+		}
+		if apierrors.IsNotFound(err) {
+			log.Info("Cluster does not exist yet, requeueing until it is created")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Could not get cluster with metadata")
+		return ctrl.Result{}, err
+	}
+
+	if annotations.IsPaused(cluster, config) {
+		log.Info("Reconciliation is paused for this object")
+		return ctrl.Result{}, nil
+	}
+
+	scope := &EKSConfigScope{
+		Logger:      log,
+		Config:      config,
+		ConfigOwner: configOwner,
+		Cluster:     cluster,
+	}
+
+	patchHelper, err := patch.NewHelper(config, r.Client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// set up defer block for updating config
+	defer func() {
+		conditions.SetSummary(config,
+			conditions.WithConditions(
+				bootstrapv1.DataSecretAvailableCondition,
+			),
+			conditions.WithStepCounter(),
+		)
+
+		patchOpts := []patch.Option{}
+		if rerr == nil {
+			patchOpts = append(patchOpts, patch.WithStatusObservedGeneration{})
+		}
+		if err := patchHelper.Patch(ctx, config, patchOpts...); err != nil {
+			log.Error(rerr, "Failed to patch config")
+			if rerr == nil {
+				rerr = err
+			}
+		}
+	}()
+
+	return r.joinWorker(ctx, scope)
+}
+
+func (r *EKSConfigReconciler) joinWorker(ctx context.Context, scope *EKSConfigScope) (ctrl.Result, error) {
+
+	if !scope.Cluster.Status.InfrastructureReady {
+		scope.Logger.Info("Cluster infrastructure is not ready, requeueing")
+		conditions.MarkFalse(scope.Config,
+			bootstrapv1.DataSecretAvailableCondition,
+			bootstrapv1.WaitingForClusterInfrastructureReason,
+			clusterv1.ConditionSeverityInfo, "")
+		return ctrl.Result{}, nil
+	}
+
+	if !scope.Cluster.Status.ControlPlaneInitialized {
+		scope.Logger.Info("Cluster has not yet been initialized, requeueing")
+		conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, "")
+		return ctrl.Result{}, nil
+	}
+
+	scope.Logger.Info("generating userdata", "cluster", scope.Cluster.GetName())
+
+	// generate userdata
+	userDataScript, err := userdata.NewNode(&userdata.NodeInput{
+		ClusterName:      scope.Cluster.GetName(),
+		KubeletExtraArgs: scope.Config.Spec.KubeletExtraArgs,
+	})
+	if err != nil {
+		scope.Error(err, "Failed to create a worker join configuration")
+		return ctrl.Result{}, err
+	}
+
+	// store userdata as secret
+	if err := r.storeBootstrapData(ctx, scope, userDataScript); err != nil {
+		scope.Error(err, "Failed to store bootstrap data")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *EKSConfigReconciler) SetupWithManager(mgr ctrl.Manager, option controller.Options) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&bootstrapv1.EKSConfig{}).
+		WithOptions(option).
+		Complete(r)
```

Comment:
For the `if !scope.Cluster.Status. ` checks above will you be adding an additional watch here on `Cluster` so that reconcile is called on an update to it?